### DISCUSSION TITLE
Task-oriented embeddings in node2vec

### DISCUSSION THREAD

**Discussion by naman-ali** [stellargraph/stellargraph#1863] [General]
Hi, how can we make the embeddings task-oriented in node2vec?

I went through the two demos you provided for node2vec, but the process of generating embedding vectors is done without considering the node_subjects(labels). Is there any way to incorporate the labels to generate better embeddings using Keras or Gensim?

**Comment by adocherty** [comment]
Hi, thanks for your question.

The random-walk based embedding methods such as node2vec are unsupervised and have no option to utilise node labels.

To generate embeddings using node labels you can use a supervised graph ML technique such as GCN or GraphSAGE and extract the implicit learned node embeddings, as detailed in the [GCN demo](https://stellargraph.readthedocs.io/en/stable/demos/node-classification/gcn-node-classification.html#Node-embeddings) or  the [GraphSAGE demo](https://stellargraph.readthedocs.io/en/stable/demos/node-classification/graphsage-inductive-node-classification.html#Node-embeddings-for-hold-out-nodes).

I hope this helps!

**Reply by naman-ali** [reply]
Yes, right. Thanks for the help. 

What you mean is that it's not possible to make it (semi)-supervised? or Stellargraph does not support it?

I was thinking that there might be a way to use the generated walks and modify Word2Vec to be supervised.
